dashdataframe/dashdf_layouts.py

```diff
@@ -0,0 +1,321 @@
+        ],
+        style={'width': '18%', 'display': 'inline-block', 'height':800, 'overflowY':'scroll', 
+                        'backgroundColor':'aliceblue', 'fontFamily':'Arial'}),
+        
+        
+        html.Div([  
+            dcc.Graph(id='indicator-graphic'),
+            
+            html.A(f'{link_name}', id='link', href='', target="_blank")
+            ],
+        
+            style={'width': '80%','float':'right', 'display':'inline-block', 'fontFamily':'Arial'}),
+
+        ]),
+    ])
+    return scatter_layout
+
+
+def make_umap_layout(plot_columns, link_name):
+    umap_layout = html.Div([
+    html.Div([
+
+        html.Div([
+            html.H1('Plot with UMAP',
+                style={'color':'purple'}),
+            dcc.Markdown(dedent('''
+            ### Explore your dataset using UMAP 
+            Note: The points represented are those visible in the scatter plot explorer 
+            For example, if you have selected a subset using Filter Sort, this representation will only show those points.
+            
+            ''')),
+
+            ],
+            style={'width': '100%', 'display': 'inline-block', 'fontFamily':'Arial'}),
+
+        html.Div([
+            dcc.Markdown(dedent('''
+            Select the features you would like to include in your UMAP plot
+            ''')),
+            dcc.Markdown(dedent('''
+            Note: If certain points are missing features you have selected, they will not be included
+            ''')),
+
+            dcc.Checklist(
+            id = 'umap_metrics',
+            options = [{'label': m, 'value':m} for m in plot_columns],
+
+            values=[],
+            labelStyle={'fontSize': '14', 'display': 'inline-block'}
+            ),
+
+            html.Button('Select All', 
+                    id='umap_features_button',
+                    style={'fontSize':'12px'}),
+
+            html.Button('Reset All', 
+                    id='unselect_button',
+                    style={'fontSize':'12px'}),
+
+            dcc.RadioItems(
+                id='umap_norm',
+                options=[{'label': i, 'value': i} for i in ['Raw_Values','Znorm']],
+                value='Raw_Values',
+                labelStyle={'fontSize':'13','display': 'inline-block'}
+                ),
+
+            html.Button('Calculate UMAP', 
+                    id='umap_button',
+                    style={'fontSize':'12px'})
+
+        ],
+        style={'width': '18%', 'display': 'inline-block', 'height':800, 'overflowY':'scroll',
+                'backgroundColor':'ghostwhite', 'fontFamily':'Arial'}),
+
+        html.Div([
+
+            dcc.Graph(id='umap-graphic'),
+
+            html.A(f'{link_name}', id='umap_link', href='', target="_blank")
+        ],
+        style={'width': '80%','float':'right', 'display':'inline-block', 'fontFamily':'Arial'}),
+
+        ]),
+    ])
+    return umap_layout
+
+def make_cluster_layout(plot_columns, link_name):
```

Comment:
along the same lines of other comments.. rather than one layout file, why not seperate modules for each tab and importing those into the main one.